File: multi-architecture-app/python-fastapi/app/controllers/greeting_controller.py
from fastapi import APIRouter, Depends

from app.models.greeting import GreetingResponse
from app.services.greeting_service import GreetingService

router = APIRouter(prefix="/hello", tags=["hello"])


def get_service() -> GreetingService:
    return GreetingService()


@router.get("/", response_model=GreetingResponse)
def read_greeting(service: GreetingService = Depends(get_service)) -> GreetingResponse:
    return service.get_greeting()

File: multi-architecture-app/node-express/src/controllers/greetingController.js
const express = require("express");
const GreetingService = require("../services/greetingService");

function createGreetingRouter(service = new GreetingService()) {
  const router = express.Router();

  router.get("/hello", (req, res) => {
    const response = service.getGreeting();
    res.json(response);
  });

  return router;
}

module.exports = createGreetingRouter;
